src/client.ts

```diff
@@ -8947,9 +8947,9 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         room: Room,
         events?: MatrixEvent[],
     ): void {
-        if (!events?.length) {
-            return;
-        }
+        if (!events?.length) return;
+        if (!room) return;
```

Comment:
L8947 says room must be provided

Reply:
Well, it feels safer to have the if statement inside the method in case something like this happens again though it feels weird to say the room can be falsey, so I am not sure which one I should go with...